### PR TITLE
clarify what command can be run where

### LINKED ISSUE #2656: make doesn't clearly distinguish where you need to be
As examples:

`make shell` only works from the local host and connects you to the docker instance.
`make test` only works from the local host and connects you to the docker instance.
`make update_code` works in either place.
`make reindex` only works from the docker instance.
`make install_python_dependencies` should only be run inside the docker instance.

Confusing.

### PR DESCRIPTION
**Pull request by andymckay**
* fixes #2656 
example:
```
addons-server make update_db                                                                          master  ✭ ✱ 
Makefile:164: *** Command should be run in the container.  Stop.
addons-server make shell                                                                              master  ✭ ✱ 
docker exec -t -i "addonsserver_web_1" bash
[root@e6b697673d91 code]# make update_db
schematic src/olympia/migrations
[root@e6b697673d91 code]# exit
```

... a check i've got all the logic right here would be appreciated

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,59 +27,95 @@ ifneq ($(NPM_CONFIG_PREFIX),)
...
-	@echo "  shell             to connect to a running addons-server docker shell"
-	@echo "  debug             to connect to a running addons-server docker for debugging"
-	@echo "  make              to connect to a running addons-server docker and run make ARGS"
-	@echo "  docs              to builds the docs for Zamboni"
-	@echo "  test              to run the entire test suite"
-	@echo "  test_force_db     to run the entire test suite with a new database"
-	@echo "  tdd               to run the entire test suite, but stop on the first error"
-	@echo "  test_failed       to rerun the failed tests from the previous run"
+	@echo "Please use 'make <target>' where <target> is one of the following commands."
+	@echo "Commands that can be run in the container:"
```
Comment (EnTeQuAk) [inline Makefile]: maybe let's say "in the docker container"?

**Comment by EnTeQuAk** [comment]
I'm r- this the way it is currently. We're using `update_deps` as our **single source of dependencies** (which was more important with pyrepo but still…) and travis is using it too.

Also, this now leaves absolutely no way to install things outside of a docker-container, whether or not we want to support it or not. Can we only show a warning but still execute the commands? That would fix travis tests too.

**Comment by andymckay** [comment]
They are now all warnings, not errors.

**Comment by EnTeQuAk** [comment]
This looks good to me now with the changes. Unless you want to add something else, lgtm 

r+

**Pull request merged**